hazelcast-spring/src/main/java/com/hazelcast/spring/AbstractHazelcastBeanDefinitionParser.java

```diff
@@ -468,50 +461,45 @@ protected void handleDiscoveryStrategies(Node node, BeanDefinitionBuilder joinCo
-            final NamedNodeMap attrs = node.getAttributes();
+        private void handleDiscoveryServiceProvider(Node node, BeanDefinitionBuilder discoveryConfigBuilder) {
+            NamedNodeMap attrs = node.getAttributes();
             Node implNode = attrs.getNamedItem("implementation");
             String implementation = implNode != null ? getTextContent(implNode) : null;
-            Assert.isTrue(implementation != null,
-                    "`implementation' attribute is required "
-                            + "to create DiscoveryServiceProvider!");
-            discoveyConfigBuilder.addPropertyReference("discoveryServiceProvider", implementation);
+            isTrue(implementation != null, "'implementation' attribute is required to create DiscoveryServiceProvider!");
+            discoveryConfigBuilder.addPropertyReference("discoveryServiceProvider", implementation);
         }
 
         private void handleDiscoveryNodeFilter(Node node, BeanDefinitionBuilder discoveryConfigBuilder) {
-            final NamedNodeMap attrs = node.getAttributes();
+            NamedNodeMap attrs = node.getAttributes();
             Node classNameNode = attrs.getNamedItem("class-name");
             String className = classNameNode != null ? getTextContent(classNameNode) : null;
             Node implNode = attrs.getNamedItem("implementation");
             String implementation = implNode != null ? getTextContent(implNode) : null;
-            Assert.isTrue(className != null || implementation != null,
-                    "One of 'class-name' or 'implementation' attributes is required "
-                            + "to create NodeFilter!");
+            isTrue(className != null || implementation != null,
+                    "One of 'class-name' or 'implementation' attributes is required to create NodeFilter!");
             discoveryConfigBuilder.addPropertyValue("nodeFilterClass", className);
             if (implementation != null) {
                 discoveryConfigBuilder.addPropertyReference("nodeFilter", implementation);
             }
         }
 
-        private void handleDiscoveryStrategy(Node node, ManagedList discoveryStrategyConfigs) {
-            final BeanDefinitionBuilder discoveryStrategyConfigBuilder =
-                    createBeanBuilder(DiscoveryStrategyConfig.class);
-            final NamedNodeMap attrs = node.getAttributes();
+        private void handleDiscoveryStrategy(Node node, ManagedList<BeanDefinition> discoveryStrategyConfigs) {
+            BeanDefinitionBuilder discoveryStrategyConfigBuilder = createBeanBuilder(DiscoveryStrategyConfig.class);
+            NamedNodeMap attrs = node.getAttributes();
             Node classNameNode = attrs.getNamedItem("class-name");
-            String className = classNameNode != null ? getTextContent(classNameNode) : null;
+            String className = getTextContent(classNameNode).trim();
```

Comment:
We use the `getTextContent(node).trim()` pattern in a lot of `handle()` methods already. The `getTextContent()` returns an empty String instead of `null`, so combined with `trim()` we can use `isEmpty()` instead of a manual `null` check.

The change was motivated by a warning from IDEA on the second `null` check on `className` (which was always failing). I also aligned the other discovery related `handle()` methods.